### ISSUE TITLE
Module '"llamaindex"' has no exported member 'Document'. (and Metadata)

### ISSUE THREAD

**Issue by zepedroc** [run-llama/LlamaIndexTS#1136]
When updating to version 0.4.6, I started having a Typescript error saying that **Document** and **Metadata** are not being exported from 'llamaindex' module. 
On version 0.4.5 everything is fine.
We're using version 5.1.6 of Typescript, in case that's important.

![image](https://github.com/user-attachments/assets/6f311fa7-9a74-4f62-b74e-f29b0cec8b26)
![image](https://github.com/user-attachments/assets/f025b29b-4243-4f2a-87cc-f2de2e301978)
![image](https://github.com/user-attachments/assets/01aaf88f-b471-4fe3-a4d3-a78e1f081ff5)

If someone could help that would be great.
We're trying to use gpt-4o-mini but I believe we need version 0.5.6 for that.

**Comment by marcusschiesser** [comment]
What's wrong with using the latest 0.5.20?

**Comment by zepedroc** [comment]
I have the same problem, only mentioned 0.4.6 so it's easier for you guys to find the potential problem.

My tsconfig.json in case it helps:
![image](https://github.com/user-attachments/assets/b4d043a7-13ee-4f0e-86ba-b88963b11a0c)

**Comment by donholly** [comment]
I am having this same issue with `0.5.20` FWIW. Haven't had much time to debug it yet, but I saw this comment in the change log: https://github.com/run-llama/LlamaIndexTS/blob/2411c9fbd0f371b0a8302327178f4e1cdda3d9f5/packages/llamaindex/CHANGELOG.md?plain=1#L795

I tried playing around with my `tsconfig.json` and updating `moduleResolution` to `bundler` or `nodenext`, but that introduced other issues with other dependencies in my project.

Happy to provide more details if helpful!

**Comment by marcusschiesser** [comment]
Yes, I remember @himself65 recommending using `bundler`. You might want to base your `tsconfig.json` on the one from the examples: https://github.com/run-llama/LlamaIndexTS/blob/main/examples/tsconfig.json

**Comment by himself65** [comment]
Yeah, please use `bundler` if you are using like next.js, vite...
use `node16` if you are using pure node.js, which means running code with `node index.js`.
